python/pyspark/sql/dataframe.py

```diff
@@ -437,10 +437,12 @@ def hint(self, name, *parameters):
         if not isinstance(name, str):
             raise TypeError("name should be provided as str, got {0}".format(type(name)))
 
+        allowed_types = (basestring, list, float, int)
```

Comment:
If we allow `list`, should we check the types recursively?

Reply:
Good catch - the Scala API accepts any, but since i don't know about the python->scala conversion, I can't answer if it is necessary to restrict to primite python types. Whatever, I get noticed that even dict works, so I extended it.